exec/repository.go

```diff
@@ -36,23 +36,29 @@ func (repo *Repository) Add(szenario *TestScenario, testgroup string, concurrenc
 }
 
 // Run all testScenarios, which match the supplied filter criterias.
-func (repo *Repository) RunTestScenarios(testgroupRegex string, nameRegex string, tagPatterns ...string) {
+func (repo *Repository) RunTestScenarios(testgroupRegex string, nameRegex string, tagPatterns ...string) map[*repositoryEntry][]*Execution {
```

Comment:
1. You should not return a private type (lower case letter on repositoryEntry) from a public interface (uppercase RunTestScenario)

2. Maybe it would be better to return an aggregated information in favour of the detailed execution structure here,
e.g. a struct with 
  - Number of total scenarios
  - Number of scenarios with errors
  - List of scenario names with errors
The details should be reported as over a reporter interfaces which e.g. writes the result to a log.